.github/workflows/operator-integration-test.yml

```diff
@@ -0,0 +1,90 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+name: operator Integration Test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [ opened, reopened ]
+    branches:
+      - main
+  workflow_dispatch:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  AmazonCloudWatchAgentOperatorIntegrationTest:
+    name: AmazonCloudWatchAgentOperatorIntegrationTest
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Start minikube
+        uses: medyagh/setup-minikube@master
+
+      - name: Deploy cert-manager to minikube
+        run:
+          kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.12.0/cert-manager.yaml
+
+      - name: Verify minikube and cert-manager
+        run: |
+          sleep 10
+          kubectl get pods -A
+
+      - name: Build image
+        run: |
+          eval $(minikube docker-env)
+          make container
+          docker images
```

Comment:
I see in the build run , below command was executed to docker build
docker buildx build --load --platform linux/amd64 -t public.ecr.aws/cloudwatch-agent/cloudwatch-agent-operator:1.0.2 --build-arg VERSION_PKG="github.com/aws/amazon-cloudwatch-agent-operator/internal/version" --build-arg VERSION=1.0.2 --build-arg VERSION_DATE=2023-12-09T05:03:05Z --build-arg AGENT_VERSION="1.300031.1b317" --build-arg AUTO_INSTRUMENTATION_JAVA_VERSION="v1.31.1" .

Any reason to build this to public ecr image?

Reply:
this is just the name of the container image after we run `make container`. When we deploy to minikube, it will take preference to the local image from `make container` which is the public ecr name. 

if this is too confusing, we can change `make container` to output something like `cloudwatch-agent-operator` and it would still work.

Reply:
Here we are not actually using the public image. We are building a new image with "make container" with public ecr name.

Reply:
I believe its better to use other name from public ecr image, it makes me assume that we are testing the public ecr image directly

Reply:
that's fair, we would have to just change the name in the Makefile and kustomize.

Reply:
changed to aws/cloudwatch-agent-operator:dev